packages/material-ui-system/src/style.js

```diff
@@ -29,10 +29,10 @@ function style(options) {
         value = themeMapping[propValueFinal];
       } else {
         value = getPath(themeMapping, propValueFinal) || propValueFinal;
+      }
 
-        if (transform) {
-          value = transform(value);
-        }
+      if (transform) {
+        value = transform(value);
```

Comment:
This sounds like a great idea. It would further unify the case where an intermediary value is collected from the theme (using the prop) and the case where the value is provided raw by the user (using the prop).

I have looked at the existing logic. It should still work correctly, the branch change potential is limited.

Reply:
I know you look into the runtime aspect and thats fine. I just want to add, before I forget it: type-wise making the transform value inferred via generic would work for all cases in which it currently works, but it would still be limiting, because the passed value (via prop on component) exactly due to the mapping in theme might have a different type, then passed after theme-mapping to the transform function. I will look into possible solution for it.